### PR TITLE
fix: 404 duplicate code and issue in

### PR DESCRIPTION
**Pull request by mayuran-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/features/components/molecules/protected-route/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import PageNotFound from 'features/pages/404-error'
+import InitialLoader from 'components/elements/dot-loader'
+
+interface ProtectedRouteProps {
+    condition: boolean
+    params?: any
```
Comment (prince-deriv) [inline src/features/components/molecules/protected-route/index.tsx:7]: it is not advisable to have `any` type. Can we pass the exact component with props instead so we don't need to create a separate `param` prop

ex:

```
  <ProtectedRoute
            is_loading={is_region_loading}
            component={<AsiansDigitalOptionsPage prop_a={1} prop_b={2} prop_c={3}  />}
            condition={is_row}
        />
```

**Review by meenakshi-deriv — APPROVED** [review]

File: src/features/components/molecules/protected-route/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import PageNotFound from 'features/pages/404-error'
+import InitialLoader from 'components/elements/dot-loader'
+
+interface ProtectedRouteProps {
+    condition: boolean
```
Comment (aswathy-deriv) [inline src/features/components/molecules/protected-route/index.tsx:6]: Can you make this boolean name as `is_condition`

Comment (prince-deriv) [reply src/features/components/molecules/protected-route/index.tsx]: `show` `is_visible`

Comment (mayuran-deriv) [reply src/features/components/molecules/protected-route/index.tsx]: updated with is_page_available @prince-deriv

**Review by hasan-deriv — APPROVED** [review]

**Review by hasan-deriv — APPROVED** [review]

**Pull request merged**